File: (4)-10.21 (Java & Python)/isPalindrome.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {
    public static void main(String args[] ) throws Exception {
      // Set up scanner and grab the basic input (pre-made setup already)
      Scanner scanner = new Scanner(System.in);
      String temp = scanner.nextLine();
      int a[] = new int[temp.length()];
      for(int i = 0; i < temp.length(); i++){
         a[i] = temp.charAt(i) - '0';
      }

      // Start your code here
      String num = "";              // to output the number at the end
      Boolean palindrome = true;    // to determine if it is or it isnt a palindrome
      
      // loop through all digits
      for(int i = 0; i < temp.length(); i++){
        num += a[i];                // append or concatenate the digit
        
        // checks if the digit from the front is NOT equal to the digit from the back  
        if(a[i] != a[temp.length() - i -1]){        
          palindrome = false;       // set palindrom (boolean) to false
        }
      }
      
      // output results
      if(palindrome){               
        System.out.println(num + " is a palindrome");
      }
      else{
        System.out.println(num + " is not a palindrome");
      }

	}
}

File: WHS Lessons/(04)-10.21 (Java & Python)/isPalindrome/isPalindrome.py
# Set up initial variables and grab the basic input
a = str(input())
number = [int(i) for i in a]
# Start your code here

# A flag that will be true if the number is a palindrome. Once the program finds out that it is not a palindrome, the flag will switch to false
flag = True

# Iterates through all the digits
for i in range(len(number)):
    if(number[i]!=number[-i-1]):
        flag = False # If nth digit from the front is not equal to the nth digit from the back, then the number is not a palindrome and flag is set to false

# Print output based on results
if(flag):
    print(a + " is a palindrome")
else:
    print(a + " is not a palindrome")